pkg/evaluator/evaluator.go

```diff
@@ -214,3 +218,97 @@ func (e *Evaluator) evalExprList(scope *scope, terms []parser.Node) []Value {
 
 	return result
 }
+
+func (e *Evaluator) evalUnaryExpr(scope *scope, expr *parser.UnaryExpression) Value {
+	right := e.Eval(scope, expr.Right)
+	if isError(right) {
+		return right
+	}
+	op := expr.Op
+	switch right := right.(type) {
+	case *Num:
+		if op == parser.OP_MINUS {
+			return &Num{Val: -right.Val}
+		}
+	case *Bool:
+		if op == parser.OP_BANG {
+			return &Bool{Val: !right.Val}
```

Comment:
should this use `boolVal()` like the comparison operators? Otherwise this creates a new Bool value and is not one of `TRUE` or `FALSE`